### ISSUE TITLE
android sleep and audio playing

### ISSUE THREAD

**Issue by plaidman** [mmatyas/pegasus-frontend#928]
### The issue

When I have audio playing, and I press power to sleep the device, the audio continues to play. I'm not sure if the audio code is keeping the device awake or if it's something else in pegasus that's keeping it awake and the audio is just a symptom, but it doesn't appear to be following the android lifecycle somewhere.

### System info

Tested on two different android devices: Retroid Pocket 2+ (Android 9), and Galaxy S9 (Android 10)

### Pegasus version

Android stable build from Dec 13th. alpha15-85-gfff1a5b2

### Pegasus log

[lastrun.log](https://github.com/mmatyas/pegasus-frontend/files/7849381/lastrun.log)

**Comment by mmatyas** [comment]
I wonder if Android thinks it's a music player and lets it running? It's certainly not something I've manually asked for, so probably comes from a lower level dependency.

**Comment by plaidman** [comment]
good point. is there a hook that you can expose in pegasus like onSleep and onWake or something that we can use to pause playback?

**Comment by mmatyas** [comment]
So far it seems I cannot, as this happens not on the theme/app level but on Android's application management level. Looks like I could only request to *not* stop the program even in sleep, as stopping it is supposed to be the default behavior.

I've also found the following: https://stackoverflow.com/a/54079128

**Comment by plaidman** [comment]
That makes sense. If the app is moved to the background or killed, then the audio will probably stop, but android doesn't want to kill the app because audio is playing. Those kinds of events wouldn't be helpful because if they occurred, then the audio would stop anyway. How about an event for screen turning on and off?

**Comment by plaidman** [comment]
fantastic, I will try this and report back.

**Comment by plaidman** [comment]
I have an update for this. I tried this code (or a bit tweaked version of it) and it works well in windows - when I focus on another window, the music stops. I will try this on android tomorrow to be sure, but I'm hopeful!

**Comment by plaidman** [comment]
Indeed it works on android! thanks guys I'll close the issue.
